protocol/custom-gas-token.md

```diff
@@ -0,0 +1,330 @@
+# Custom Gas Token: Design Doc
+
+| Author | *AgusDuha, Joxes, Skeletor* |
+| --- | --- |
+| Created at | *2025-07-24* |
+| Initial Reviewers | *Mark Tyneway, TBD* |
+| Need Approval From | *TBD* |
+| Status | *In Review* |
+
+## Purpose
+
+Enable OP Stack chains to use an asset other than ETH as their native fee currency by introducing a minimal, flexible, and standardizable Custom Gas Token (CGT) architecture. 
+
+## Summary
+
+The proposed Custom Gas Token upgrade lets any OP Stack chain introduce its native asset as the gas currency with almost no core-code intrusion: a single `isCustomGasToken()` flag turns off ETH transfer flows in all bridging methods, while three new pre-deploys, `NativeAssetLiquidity`, a contract with pre-minted assets, `LiquidityController`, an owner-governed mint/burn router, and `L2CGTBridge`, an reserved address that contains a contract with the API calls, hand supply control to chain governors or authorized “minter” contracts that can plug in anything from ERC-20 converters to third-party bridgind models or emission schedules. Wrapped-asset compatibility is preserved, and the entire system launches at genesis by funding the vault and letting the governor release an initial working balance. Overall, the design keeps the OP Stack lean, token-agnostic, and future-proof while unlocking custom economics and flexibility to make the native asset expressive in its manner.
```

Comment:
What do you think of merging LiquidityController and NativeAssetLiquidity? We have a lot of contracts, and these two are fairly small and logically related, so combining them might be simpler